### PR TITLE
Normalize sorting options

### PR DESCRIPTION
**Pull request by billylanchantin**
Closes:

* https://github.com/elixir-explorer/explorer/issues/765

Each of these options are now available on all the sorting functions:

* `direction: :asc | :desc`
  * This takes the form of keys in the first arg on the DF functions
* `nils: :first | :last`
* `parallel: true | false`
  * This isn't available on the lazy functions
* `stable: true | false`

### CODE REVIEW

File: lib/explorer/data_frame.ex [hunk 1]
```diff
@@ -3052,14 +3052,34 @@ defmodule Explorer.DataFrame do
...
 
-    * `:stable` - `boolean()`.
+    * `:parallel` - boolean.
+      Whether to parallelize the sorting.
+      By default it is `false`.
+
+      > #### Notice {: .notice}
+      >
+      > The `:parallel` option is applicable under one condition: all lazy series must reduce to
+      > `op: :column`. E.g.
```
Comment (billylanchantin) [inline lib/explorer/data_frame.ex:3065]: I had trouble explaining/wording this notice.

File: lib/explorer/polars_backend/data_frame.ex [hunk 2]
```diff
@@ -643,22 +643,44 @@ defmodule Explorer.PolarsBackend.DataFrame do
...
+        |> Enum.unzip()
+
+      Shared.apply_dataframe(df, out_df, :df_arrange, [
+        column_names,
+        directions,
+        maintain_order?,
+        multithreaded?,
+        nulls_last?,
+        df.groups
+      ])
```
Comment (billylanchantin) [inline lib/explorer/polars_backend/data_frame.ex:669]: We weren't ever actually calling `df_arrange` from the Elixir side. You can think of this clause as basically an optimization.

File: lib/explorer/shared.ex [hunk 3]
```diff
@@ -475,4 +475,58 @@ defmodule Explorer.Shared do
     |> Enum.sort(&(elem(&1, 1) <= elem(&2, 1)))
     |> Enum.map(fn {_, key} -> ["      * ", inspect(key), ?\n] end)
   end
+
+  @doc """
+  Validate options to be used for the various sorting functions.
+  """
+  def validate_sort_options!(opts) do
```
Comment (billylanchantin) [inline lib/explorer/shared.ex:482]: Someone might wanna double check my logic on this one 😅

File: lib/explorer/data_frame.ex (hunk 1 above)
Comment (philss) [reply lib/explorer/data_frame.ex:3065]: I like this way with the example!

**Review by philss — APPROVED** [review]
LGTM!

**Review by josevalim — COMMENTED** [review]
Beautiful work. The only suggestion I have is about the defaults.

Numpy and Pandas do not use stable sorting by default nor Polars. Polars is also parallel by default, which I believe we should try to be as well. So my suggestion is to make stable default to false and parallel default to true. This way, we can also simplify the explanation of parallel. We could say "Use parallel algorithms when possible. Set it to false to disable it.".

Thoughts?

**Comment by billylanchantin** [comment]
@josevalim That sounds great. I was keeping the defaults we had, but I hadn't thought about what they _ought_ to be. I can definitely have `:stable` default to `false` and `:parallel` default to `true`.

> This way, we can also simplify the explanation of parallel. We could say "Use parallel algorithms when possible. Set it to false to disable it.".

That's a good tactic: describe the philosophy instead of belaboring the implementation (which may change). I think I'll still add a sentence about parallelism not always being available on lazy operations. That way the reader will have some idea of what we mean by "when possible".

File: lib/explorer/data_frame.ex (hunk 1 above)
Comment (billylanchantin) [reply lib/explorer/data_frame.ex]: Thanks! I ended up dropping it for simpler wording: [3c77fe5](https://github.com/elixir-explorer/explorer/pull/771/commits/3c77fe564f05a0f12a448e1e83ace922537259a6)

But if enough users say they're confused, we can resurrect that notice.

**Review by josevalim — APPROVED** [review]

**Pull request merged**